core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -265,25 +265,24 @@ class KafkaApis(val requestChannel: RequestChannel,
       sendResponseMaybeThrottle(request, requestThrottleMs => new OffsetCommitResponse(requestThrottleMs, results.asJava))
     } else {
 
-      var unauthorizedTopics = Set[TopicPartition]()
-      var nonExistingTopics = Set[TopicPartition]()
-      var authorizedTopics = mutable.Map[TopicPartition, OffsetCommitRequest.PartitionData]()
+      val unauthorizedTopicErrors = mutable.Map[TopicPartition, Errors]()
+      val nonExistingTopicErrors = mutable.Map[TopicPartition, Errors]()
+      val authorizedTopicRequestInfoBldr = immutable.Map.newBuilder[TopicPartition, OffsetCommitRequest.PartitionData]
```

Comment:
Why not just use a mutable map?

Reply:
Oh, just because the types below already expected an immutable map. Would you suggest changing that or going with the copy?

Reply:
What you have is fine, I think.